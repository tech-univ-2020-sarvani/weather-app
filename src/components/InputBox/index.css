.input{
    width: 50%;
    padding: 12px 20px;
    margin-top: 150px;
    display: inline-block;
    border: 2px solid rgb(20, 19, 19);
    border-radius: 4px;
    box-sizing: border-box;
  }