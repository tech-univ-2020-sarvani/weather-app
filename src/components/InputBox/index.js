import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.css';

const InputBox = (props) => {
  const { testId, setText } = props;
  return (
    <input className="input" type="text" data-testid={testId} onChange={(e) => setText(e.target.value)} />
  );
};

InputBox.propTypes = {
  testId: PropTypes.string,
  setText: PropTypes.func.isRequired,
};

export default InputBox;
