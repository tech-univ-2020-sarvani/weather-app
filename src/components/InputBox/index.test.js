import React from 'react';
import { render } from '@testing-library/react';
import InputBox from './index';

describe('The InputBox component', () => {
  it('shouls render correctly', () => {
    const { asFragment } = render(<InputBox testId="test-input" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
