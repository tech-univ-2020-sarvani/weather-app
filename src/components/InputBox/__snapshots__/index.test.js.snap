// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The InputBox component shouls render correctly 1`] = `
<DocumentFragment>
  <input
    data-testid="test-input"
    type="text"
  />
</DocumentFragment>
`;
